copyoptions.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oras
+
+import (
+	"context"
+	"math"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"golang.org/x/sync/semaphore"
+)
+
+const defaultConcurrencyLimit = 3

```

Comment:
Leart this from [containerd](https://sourcegraph.com/github.com/containerd/containerd@eb39aac/-/blob/pkg/cri/config/config_unix.go?L101) and [moby](https://sourcegraph.com/github.com/moby/moby@8abb84c/-/blob/daemon/config/config.go?L26)